src/CalcManager/CEngine/scioper.cpp

```diff
@@ -112,8 +112,17 @@ CalcEngine::Rational CCalcEngine::DoOperation(int operation, CalcEngine::Rationa
             }
             else
             {
-                iFinalSign = iNumeratorSign;
-                result %= temp;
+                if (m_fIntegerMode)
+                {
+                    // Programmer mode
+                    iFinalSign = iNumeratorSign;
+                    result %= temp;
+                }
+                else
+                {
+                    iFinalSign = iDenominatorSign;
+                    result = Mod(result, temp);
```

Comment:
Let's throw a comment on here as well that this is the arithmetic mod. The more clear we can make the code distinguishing our use of %/remrat and Mod/modrat and noting that the decision to use one or the other is deliberate, the better.